sdk/core/Azure.Core/src/Pipeline/HttpPipelineMessage.cs

```diff
@@ -60,5 +60,76 @@ public void SetProperty(string name, object value)
 
             _properties[name] = value;
         }
+
+        public Stream? PreserveResponseContent()
+        {
+            switch (_response?.ContentStream)
+            {
+                case ResponseShouldNotBeUsedStream responseContent:
+                    return responseContent.Original;
+                case Stream stream :
+                    _response.ContentStream = new ResponseShouldNotBeUsedStream(_response.ContentStream);
+                    return stream;
+                default:
+                    return null;
+            }
+        }
+
+        public void Dispose()
+        {
+            Request?.Dispose();
+            _response?.Dispose();
+        }
+
+        private class ResponseShouldNotBeUsedStream: Stream
+        {
+            public Stream Original { get; }
+
+            public ResponseShouldNotBeUsedStream(Stream original)
+            {
+                Original = original;
+            }
+
+            private static Exception CreateException()
+            {
+                return new InvalidOperationException("This operation returns Stream as part of the model type. It should be used instead of the response content stream.");
+            }
+
+            public override void Flush()
+            {
+                throw CreateException();
+            }
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                throw CreateException();
+            }
+
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                throw CreateException();
+            }
+
+            public override void SetLength(long value)
+            {
+                throw CreateException();
+            }
+
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                throw CreateException();
+            }
+
+            public override bool CanRead => throw CreateException();
```

Comment:
Nit - maybe do this for all the methods too?

Reply:
I think others would eventually call into one of these ones